llama.h

```diff
@@ -641,6 +641,10 @@ extern "C" {
     // n_threads_batch is the number of threads used for prompt and batch processing (multiple tokens)
     LLAMA_API void llama_set_n_threads(struct llama_context * ctx, uint32_t n_threads, uint32_t n_threads_batch);
 
+    // Set whether to use causal attention or not
+    // If set to true, the model will only attend to the past tokens
+    LLAMA_API void llama_set_embeddings(struct llama_context * ctx, bool embeddings);
```

Comment:
This should be improved - the comment is about causal attention, but the function is called `llama_set_embeddings`. It's not clear enough

I think we need to introduce `bool causal_attn` in `llama_cparams`. It will be initialized by default with `llama_hparams.causal_attn`, but it would be possible to override it via `llama_set_causal_attn(struct llama_context * ctx, bool causal_attn);`

The `if` in `llama_set_inputs()` should be checking `cparams.causal_attn`. It would make more sense, because the `if` is about whether we build a causal mask or not. But the proposed change `if (!cparams.embeddings) {` is confusing

Reply:
Agreed, that is confusing. Pushing proposed changes now.